charts/frontend/templates/services-cron.yaml

```diff
@@ -43,13 +43,59 @@ spec:
             - name: {{ $key }}
               value: {{ $val | quote }}
             {{- end }}
-            command: ["/bin/sh", "-c"]
+            command: ["/bin/bash", "-c"]
             args:
               - |
-                 set -ex
-                 echo "starting cron run"
-                 {{ $job.command | nindent 18 }}
-                 echo "cron run completed"
+                # We need to save the executable command(s) into a file,
+                # because the 'time' can execute only command files,
+                # not the shell constructs.
+                script="/tmp/cron.$$"
+                cat << CMD > "${script}"
+                #!/bin/bash
+                set -x
+                {{ $job.command | nindent 16 }}
+                CMD
+                chmod 755 "${script}"
+
+                # Try to make the temporary files more unique.
+                # $$ = PID of the current running shell.
+                stats="/tmp/stats.$$"
+                output="/tmp/output.$$"
+                output_json="/tmp/output_json.$$"
+
+                # For backward compatibility.
+                echo "starting cron run"
+
+                # The execution of the actual cron command.
+                # For backward compatibility reasons send the output to stdout as well (tee).
+                /usr/bin/time -f '"duration_s":%e,"peak_mem_kb":%M' -o "${stats}" "${script}" 2>&1 \
+                | tee "${output}"
+
+                # For backward compatibility.
+                echo "cron run completed"
+
+                # It's tricky to get the exit code of a random command from a pipeline.
+                # It is highly shell-dependent.
+                exit_code="${PIPESTATUS[0]}"
+
+                # Convert the command output to JSON.
+                node -e 'const fs = require("fs"); console.log(JSON.stringify(fs.readFileSync(0, "utf-8")));' < "${output}" > "${output_json}"
```

Comment:
Here I need to turn to @Jancis or @tharna with my question:

Can we _safely assume_ that `node` and `bash` (line 46) are always present in the services? Technically I guess it is possible to use the frontend chart but define a Dockerfile that does not use our silta-node base image.

Reply:
Correct, `node` might not be absent. There are services that run a different base image